packages/modal/src/Modal/Modal.tsx

```diff
@@ -45,18 +47,30 @@ const Root = ({
   return <RootComponent open={open} onOpenChange={onOpenChange} {...props} />;
 };
 
+type RootComponentProps = RootProps & {
+  open: boolean;
+  onOpenChange: (value: boolean) => void;
+};
+
 const RootComponent = ({
   open,
   onOpenChange,
   a11yTitle,
   a11yDescription,
   children,
   layer: layerProp,
+  // Focus scope props
+  trapped,
+  onMountAutoFocus,
+  onUnmountAutoFocus,
   ...props
-}: Required<RootProps, "open" | "onOpenChange">) => {
-  const id = React.useId();
+}: RootComponentProps) => {
+  // We use the a11yTitle as the id for the modal as it is unique
+  // and can be used to identify the modal in the stacking context.
+  // Without the need to generate a random id and manage between
+  // different modal rendering patterns.
+  const id = a11yTitle;
```

Comment:
I originally used `React.useId()` here. This worked fine when modals were "nested" side by side, i.e.

```
<Modal></Modal>
<Modal></Modal>
<Modal></Modal>
```

But when actually nesting modals inside of each other like this:

```
<Modal>
   <Modal>
      <Modal>
      </Modal>
    </Modal>
</Modal
```

The id generated from `useId()` would update on re-render. This meant that there was no consistent key that we could use to manage the layers within the stacking context. 

Fortunately, the `a11yTitle` prop is a required prop that is meant to be unique. This means that we can have a consistent key that works outside of re-renders. So making this change fixed some of the modal stacking issues I was seeing after utilizing a portal.